### PR TITLE
CRED-347 Make Chrome run constantly as a separate instance

### PR DESCRIPTION
**Pull request by envil**
https://credinord.atlassian.net/browse/CRED-347

**Reasons for PR**:

- We decided not to continue with this approach since it's not much different
from the existing approach (due to the way puppeteer spawn a real headless
Chrome instance under the hood).
- We still want to recover from crashes

**PR Change List**:
- Improved the code for recovering from crash case

### CODE REVIEW

File: src/internal/pdf/browser.provider.ts [hunk 1]
```diff
@@ -40,10 +40,16 @@ export class BrowserProvider {
   }
 
   private getExistingBrowserInstance() {
-    return of(this.browserInstance).pipe(
-      tap(() => {
-        this.logger.debug('Returning existing browser instance');
-      }),
+    return iif(
```
Comment (zun-credinord) [inline src/internal/pdf/browser.provider.ts:43]: alternatively, should we always spawn a browser instance when the service is initialized, idk which is neatier. This current solution is fine as well.

**Review by zun-credinord — APPROVED** [review]
GJ!

File: src/internal/pdf/browser.provider.ts (hunk 1 above)
Comment (envil) [reply src/internal/pdf/browser.provider.ts:43]: We do, this's for the case the running browser instance is crashed and disconnected.

Comment (zun-credinord) [reply src/internal/pdf/browser.provider.ts:43]: this is good enough because we cover the recover here as well.

**Review by rgalimskii-cn — APPROVED** [review]
Good work! Since separately running Chrome instance turned out to be a pain in the ass we should consider paying for something like https://docs.browserless.io/ in the future.

**Pull request merged**